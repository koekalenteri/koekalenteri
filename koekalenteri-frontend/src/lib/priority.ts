export interface Priority {
  group: string
  name: string
  value: string
}

export const PRIORITY: Priority[] = [
  { group: 'Järjestävän yhdistyksen jäsen', name: 'Jäsenet', value: 'member' },
  { group: 'Etusija nimetyillä roduilla', name: 'Kultaisetnoutajat', value: '111' },
  { group: 'Etusija nimetyillä roduilla', name: 'Labradorit', value: '122' },
  { group: 'Etusija nimetyillä roduilla', name: 'Sileäkarvaiset noutajat', value: '121' },
  { group: 'Etusija nimetyillä roduilla', name: 'Chesapeakenlahdennoutajat', value: '263' },
  { group: 'Etusija nimetyillä roduilla', name: 'Novascotiannoutajat', value: '312' },
  { group: 'Etusija nimetyillä roduilla', name: 'Kiharakarvaiset noutajat', value: '110' },
  { group: 'Järjestäjän kutsumat koirat', name: 'Kutsutut', value: 'invited' },
]
