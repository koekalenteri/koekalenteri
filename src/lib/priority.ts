import { unique } from './utils'

export interface Priority {
  group: string
  name: string
  value: string
}

export const PRIORITY: Priority[] = [
  { group: 'Järjestävän yhdistyksen jäsen', name: 'Jäsenet', value: 'member' },
  { group: 'Etusija nimetyillä roduilla', name: 'Kultaisetnoutajat', value: '111' },
  { group: 'Etusija nimetyillä roduilla', name: 'Labradorit', value: '122' },
  { group: 'Etusija nimetyillä roduilla', name: 'Sileäkarvaiset noutajat', value: '121' },
  { group: 'Etusija nimetyillä roduilla', name: 'Chesapeakenlahdennoutajat', value: '263' },
  { group: 'Etusija nimetyillä roduilla', name: 'Novascotiannoutajat', value: '312' },
  { group: 'Etusija nimetyillä roduilla', name: 'Kiharakarvaiset noutajat', value: '110' },
  { group: 'Järjestäjän kutsumat koirat', name: 'Kutsutut', value: 'invited' },
]

export const priorityValuesToPriority = (values: Array<string | undefined> = []) => {
  const result: Priority[] = []
  if (!Array.isArray(values)) return result

  for (const value of unique(values)) {
    const priority = PRIORITY.find((p) => p.value === value)
    if (priority) {
      result.push(priority)
    }
  }
  return result
}
