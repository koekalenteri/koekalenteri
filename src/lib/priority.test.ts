import { PRIORITY, priorityValuesToPriority } from './priority'

describe('lib/priority', () => {
  describe('PRIORITY', () => {
    it('should match snapshot and length', () => {
      expect(PRIORITY).toMatchSnapshot()
      expect(PRIORITY.length).toEqual(8)
    })
  })

  describe('priorityValuesToPriority', () => {
    it.each`
      values
      ${undefined}
      ${null}
      ${NaN}
      ${''}
      ${{}}
      ${[null]}
      ${['kissa', 'koira']}
    `('should ignore invalid values: $values', ({ values }) => {
      expect(priorityValuesToPriority(values)).toEqual([])
    })

    it.each`
      values                                               | expected
      ${[PRIORITY[0].value]}                               | ${[PRIORITY[0]]}
      ${[PRIORITY[1].value]}                               | ${[PRIORITY[1]]}
      ${[PRIORITY[2].value, PRIORITY[3].value]}            | ${[PRIORITY[2], PRIORITY[3]]}
      ${[PRIORITY[4].value, 'kissa', PRIORITY[5].value]}   | ${[PRIORITY[4], PRIORITY[5]]}
      ${[PRIORITY[7].value, undefined, PRIORITY[7].value]} | ${[PRIORITY[7]]}
    `('should map valid values, ignoring invalid and removing duplicates : $values', ({ values, expected }) => {
      expect(priorityValuesToPriority(values)).toEqual(expected)
    })
  })
})
