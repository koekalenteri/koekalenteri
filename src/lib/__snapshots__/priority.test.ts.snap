// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/priority PRIORITY should match snapshot and length 1`] = `
[
  {
    "group": "Järjestävän yhdistyksen jäsen",
    "name": "Jäsenet",
    "value": "member",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Kultaisetnoutajat",
    "value": "111",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Labradorit",
    "value": "122",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Sileäkarvaiset noutajat",
    "value": "121",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Chesapeakenlahdennoutajat",
    "value": "263",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Novascotiannoutajat",
    "value": "312",
  },
  {
    "group": "Etusija nimetyillä roduilla",
    "name": "Kiharakarvaiset noutajat",
    "value": "110",
  },
  {
    "group": "Järjestäjän kutsumat koirat",
    "name": "Kutsutut",
    "value": "invited",
  },
]
`;
